apps/condo/domains/banking/schema/BankAccount.js

```diff
@@ -0,0 +1,195 @@
+const { validateNumber } = require('@condo/domains/banking/utils/validate/number.utils')
+
+
+const BankAccount = new GQLListSchema('BankAccount', {
+    schemaDoc: 'Bank account, that will have transactions, pulled from various integrated data sources. Transactions will be marked by categories of incomes and expenses.',
+    fields: {
+        organization: ORGANIZATION_OWNED_FIELD,
+
+        tin: {
+            schemaDoc: 'Tax Identification Number',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                validateInput: ({ existingItem, resolvedData, addFieldValidationError }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+
+                    const country = get(newItem, 'country')
+                    const tin = get(newItem, 'tin')
+
+                    const { result, errors } = validateTin(tin, country)
+
+                    if ( !result ) {
+                        addFieldValidationError(errors[0])
+                    }
+                },
+            },
+        },
+
+        country: {
+            schemaDoc: 'Country where the bank is located',
+            isRequired: true,
+            type: Select,
+            options: Object.keys(COUNTRIES).join(','),
+        },
+
+        routingNumber: {
+            schemaDoc: 'The routing transit number for the bank account.',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                validateInput: ({ existingItem, resolvedData, addFieldValidationError }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+
+                    const country = get(newItem, 'country')
+                    const routingNumber = get(newItem, 'routingNumber')
+
+                    const { result, errors } = validateRoutingNumber(routingNumber, country)
+
+                    if ( !result ) {
+                        addFieldValidationError(errors[0])
+                    }
+                },
+            },
+        },
+
+        number: {
+            schemaDoc: 'Bank account number',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                validateInput: ({ existingItem, resolvedData, addFieldValidationError }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+
+                    const country = get(newItem, 'country')
+                    const number = get(newItem, 'number')
+                    const routingNumber = get(newItem, 'routingNumber')
+
+                    const { result, errors } = validateNumber(number, routingNumber, country)
+
+                    if ( !result ) {
+                        addFieldValidationError(errors[0])
+                    }
+                },
+            },
+        },
+
+        currencyCode: CURRENCY_CODE_FIELD,
+
+        approvedAt: {
+            schemaDoc: 'When the bank account received the status of approved',
```

Comment:
It's not clear. Who needs to approve this? and why?

Reply:
Support approves bank account based on information about this organization. This process is already live in BillingRecipient model